packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -213,8 +215,41 @@ class SnackBar extends StatelessWidget {
     }
     final CurvedAnimation heightAnimation = new CurvedAnimation(parent: animation, curve: _snackBarHeightCurve);
     final CurvedAnimation fadeAnimation = new CurvedAnimation(parent: animation, curve: _snackBarFadeCurve, reverseCurve: const Threshold(0.0));
+    Widget snackbar = new SafeArea(
+      top: false,
+      child: new Row(
+        children: children,
+        crossAxisAlignment: CrossAxisAlignment.center,
+      ),
+    );
+    snackbar = new Semantics(
+      container: true,
+      liveRegion: true,
+      onDismiss: () {
+        Scaffold.of(context).removeCurrentSnackBar(reason: SnackBarClosedReason.swipe);
```

Comment:
we should either add a new SnackBarClosedReason value, or we should update the "swipe" value's documentation to cover this case.